### PR TITLE
petsc: a workaround for macOS Clang 8.1.0 and XCode 8.3

### PR DESCRIPTION
**Pull request by davydden**
unfortunately new XCode 8.3 breaks things 😢 see http://lists.mcs.anl.gov/pipermail/petsc-users/2017-March/032049.html

### CODE REVIEW

File: var/spack/repos/builtin/packages/petsc/package.py [hunk 1]
```diff
@@ -74,6 +74,13 @@ class Petsc(Package):
     variant('int64', default=False,
             description='Compile with 64bit indices')
 
+    # temporary workaround Clang 8.1.0 with XCode 8.3 on macOS, see
+    # https://bitbucket.org/petsc/petsc/commits/4f290403fdd060d09d5cb07345cbfd52670e3cbc
+    # the patch is an adoptation of the original commit to 3.7.5
+    if sys.platform == "darwin":
+        patch('macos-clang-8.1.0.diff',
+              when='@3.7.5')
```
Comment (davydden) [inline var/spack/repos/builtin/packages/petsc/package.py]: probably i need to add here `%clang@8.1.0:`, is it the way this works?

File: var/spack/repos/builtin/packages/petsc/package.py [hunk 2]
```diff
@@ -74,6 +74,13 @@ class Petsc(Package):
     variant('int64', default=False,
             description='Compile with 64bit indices')
 
+    # temporary workaround Clang 8.1.0 with XCode 8.3 on macOS, see
+    # https://bitbucket.org/petsc/petsc/commits/4f290403fdd060d09d5cb07345cbfd52670e3cbc
+    # the patch is an adoptation of the original commit to 3.7.5
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/petsc/package.py]: typo: adoptation -> adaptation

File: var/spack/repos/builtin/packages/petsc/package.py (hunk 1 above)
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/petsc/package.py]: Yep, that should work.

**Comment by BarrySmith** [comment]
Yes, this is fine

**Pull request merged**